plasmapy/particles/decorators.py

```diff
@@ -413,21 +413,22 @@ def verify_particle_name_criteria(self, parameter, particle):
         categorization criteria.
         """
 
+        if (
+            parameter == "ion"
+            and isinstance(particle, CustomParticle)
+            and particle.charge.value > 0
+            and particle.mass.value > 0
+        ):
+            return
+
         name_categorization_exception = [
             ("element", {"require": "element"}, InvalidElementError),
             ("isotope", {"require": "isotope"}, InvalidIsotopeError),
-            # TODO: In the future, "ion" should be changed to allow only
-            # ions and not neutral atoms with an explicit charge.
```

Comment:
Right now, I'm thinking that it would be fine to have `ion` be a catch-all and include neutral atoms because...

 - It makes the code simpler
 - We'd could avoid breaking changes
 - It's really often helpful to have functions work on neutrals, like `gyroradius` returning `np.inf * u.m`.

Reply:
I agree - 'atom' definitely implies neutral to me!